src/terraform-aws/content.md

```diff
@@ -0,0 +1,713 @@
+
+> If you already have Terraform installed, make sure you have at least Terraform v0.12 since this is the minimum version required by the module.
+
+Terraform is distributed as a statically linked, precompiled binary (written in Go) and to install it, you simply have to download it and move it to any directory in your `PATH`.
+
+The [Terraform website](https://www.terraform.io/downloads.html) contains the latest versions of the Terraform binaries for all supported target platforms:
+
+![Install Terraform](assets/terraform-download.png)
+
+Download the package corresponding to your platform, unzip it, and move the contained binary to some directory in your `PATH`, such as `/usr/bin`.
+
+On macOS, you can alternatively install Terraform with:
+
+```terminal|command=1|title=bash
+brew install terraform
+```
+
+Once Terraform is installed, you can verify the installation with:
+
+```terminal|command=1|title=bash
+terraform version
+```
+
+> Please make sure that you have at least version 0.12 since this is the minimum version required by the Terraform kubeadm module.
+
+### 2. Configure AWS credentials
+
+First of all, you need an [AWS account](https://aws.amazon.com/).
+
+Since the Terraform kubeadm module will create AWS resources on your behalf, it needs to have access to the [Access Key ID and Secret Access Key](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) of an [IAM User](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html) in your AWS account.
+
+If you're already using the [AWS CLI](https://aws.amazon.com/cli/) and have an `~/.aws/credentials` file on your machine, then you should be good to go.
+
+In all other cases, you can set the following environment variables:
+
+```terminal|command=1,2|title=bash
+export AWS_ACCESS_KEY_ID=<AccessKeyID>
+export AWS_SECRET_ACCESS_KEY=<SecretAccessKey>
+```
+
+You can find out the Access Key ID and Secret Access Key of all the IAM users in your AWS account in the [AWS IAM Console](https://console.aws.amazon.com/iam/home) 
+
+> In the AWS IAM Console, you can also generate a new Access Key ID and Secret Access Key for any IAM user if you can't retrieve the existing one.
+
+If you're using an IAM user with limited permissions, make sure that it has at least the following [IAM Policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies.html) attached:
+
+- `AmazonEC2FullAccess`
+- `AmazonVPCFullAccess`
+
+### 3. Ensure default OpenSSH keys
+
+The Terraform kubeadm module will set up SSH access to the nodes of the cluster by using the default key pair of OpenSSH:
+
+- `~/.ssh/id_rsa` (private key)
+- `~/.ssh/id_rsa.pub` (public key)
+
+If you currently don't have these files on your local machine, you can generate them with the following command:
+
+```terminal|command=1|title=bash
+ssh-keygen
+```
+
+_You're now ready to use the Terraform kubeadm module!_
+
+## Creating a minimal cluster
```

Comment:
I think there's a section missing... the first iteration of the module that you showed me was really small (with very little features). I think that's valuable. As a reader, I'd like to know what's going on the module. Also, can I build it myself? what's the secret sauce?

We don't have to explain everything but empower the reader to understand.